src/vunnel/tool/fixdate/grypedb.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import os
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+
+import oras.client
+import sqlalchemy as db
+
+from vunnel import workspace
+
+from .finder import Finder, Result
+
+
+@dataclass
+class FixDate:
+    vuln_id: str
+    provider: str
+    package_name: str
+    full_cpe: str
+    ecosystem: str
+    fix_version: str | None
+    first_observed_date: datetime
+    resolution: str
+    source: str
+
+
+class Store(Finder):
+    def __init__(self, ws: workspace.Workspace, provider: str) -> None:
+        self.workspace = ws
+        self.provider = provider
+        self.db_path = Path(ws.input_path) / "fix-dates" / f"{provider}.db"
+        self.logger = logging.getLogger("fixes-" + provider)
+        self.engine: db.engine.Engine | None = None
+        self.conn: db.engine.Connection | None = None
+        self.table: db.Table | None = None
+
+    async def setup(self) -> None:
```

Comment:
Can you help me understand why this is async? (I'm new to sqlalchemy 2).

Reply:
it really shouldn't be, this one got by me